cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewManager.java

```diff
@@ -66,10 +66,10 @@
   /**
    * Get the data associated with the preview.
    * @param preview the id associated with the preview
-   * @return the {@link Map} of tracer name to properties associated with the tracer for a given preview
-   * @throws NotFoundException if the previewId is not found
+   * @param tracerName the name of the tracer used for preview
+   * @return the {@link Map} of properties associated with the tracer for a given preview
    */
-  Map<String, Map<String, List<Object>>> getData(ApplicationId preview) throws NotFoundException;
+  Map<String, List<JsonElement>> getData(ApplicationId preview, String tracerName);
```

Comment:
You never need to get objects of all tracers? Do you have a method to list all tracers?